File: src/main/java/uk/gov/justice/hmpps/prison/api/model/v1/Transaction.java
package uk.gov.justice.hmpps.prison.api.model.v1;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

import jakarta.validation.constraints.NotNull;

@Schema(description = "Transaction Response")
@Data
@AllArgsConstructor
@Builder
@EqualsAndHashCode
@ToString
public class Transaction {
    @Schema(description = "ID of created transaction", example = "6179604-1")
    @NotNull
    private String id;
}

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/v1/Transaction.kt
package uk.gov.justice.hmpps.prison.api.model.v1

import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Transaction Response")
data class Transaction(
  @Schema(description = "ID of created transaction", example = "6179604-1")
  val id: String,
)
